spec/system/admin_customize_components_config_area_spec.rb

```diff
@@ -0,0 +1,332 @@
+# frozen_string_literal: true
+
+describe "Admin Customize Themes Config Area Page", type: :system do
+  fab!(:admin)
+  fab!(:parent_theme) { Fabricate(:theme, name: "A theme") }
+  fab!(:parent_theme_2) { Fabricate(:theme, name: "B theme") }
+  fab!(:parent_theme_3) { Fabricate(:theme, name: "C theme") }
+  fab!(:parent_theme_4) { Fabricate(:theme, name: "D theme") }
+
+  fab!(:enabled_component) do
+    Fabricate(
+      :theme,
+      name: "Glorious component",
+      component: true,
+      enabled: true,
+      parent_themes: [parent_theme, parent_theme_2, parent_theme_3, parent_theme_4],
+    )
+  end
+  fab!(:disabled_component) do
+    Fabricate(:theme, name: "Glossy component", component: true, enabled: false)
+  end
+  fab!(:remote_component) do
+    Fabricate(
+      :theme,
+      component: true,
+      enabled: false,
+      parent_themes: [parent_theme_3],
+      remote_theme:
+        RemoteTheme.create!(
+          remote_url: "https://github.com/discourse/tc-1.git",
+          authors: "CDCK Inc.",
+        ),
+      theme_fields: [
+        ThemeField.new(
+          name: "en",
+          type_id: ThemeField.types[:yaml],
+          target_id: Theme.targets[:translations],
+          value: <<~YAML,
+            en:
+              theme_metadata:
+                description: "Description of my remote component"
+          YAML
+        ),
+      ],
+    )
+  end
+  fab!(:remote_component_with_update) do
+    Fabricate(
+      :theme,
+      component: true,
+      enabled: false,
+      remote_theme:
+        RemoteTheme.create!(remote_url: "https://github.com/discourse/tc-2", commits_behind: 4),
+    )
+  end
+
+  let(:config_area) { PageObjects::Pages::AdminCustomizeComponentsConfigArea.new }
+  let(:toasts) { PageObjects::Components::Toasts.new }
+  let(:dialog) { PageObjects::Components::Dialog.new }
+
+  before { sign_in(admin) }
+
+  it "doesn't display filters when there are no components installed" do
+    Theme.where(component: true).destroy_all
```

Comment:
Creating components with fab! and then deleting them seems like a bit of waste. What would you say about creating 2 context blocks, one "with components" and one "without components" ?